### PR TITLE
Run events controller as separate binary

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

This is the first step towards moving the whole cloudevents logic to a dedicated controller.

The `Run` controller is already separated from the other controllers, but it's compiled and deployed in the shared binary. In this change we move that controller to a dedicated binary, with its own deployment, service account, roles and bindings.

This new binary shares the config maps from the pipeline binary, so that existing configuration options and docs continue to apply with no change.

Because of injection, all the informers are setup for this binary, which means that the service account requires read access to the various tekton resources. This is fine however considering that eventually this controller will handle events for all tekton pipeline resources.

The publish task is amended to expect the new events image by default as well.

Partially-fixes: #2944
Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
The cloudevents controller for `Run` has been moved to its own binary, with dedicated deployment, service, pod, service account, roles and role bindings. No functional change, no configuration change.
```

/kind misc

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: config/200-serviceaccount.yaml [hunk 1]
```diff
@@ -30,3 +30,13 @@ metadata:
...
     app.kubernetes.io/instance: default
     app.kubernetes.io/part-of: tekton-pipelines
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tekton-events-controller
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/component: controller
```
Comment (wlynch) [inline config/200-serviceaccount.yaml:40]:
```suggestion
    app.kubernetes.io/component: events-controller
```

Comment (wlynch) [reply config/200-serviceaccount.yaml:40]: Or it looks like you're using `events` elsewhere. No preference, but we should be consistent and not use the existing `controller` component.

File: config/201-rolebinding.yaml [hunk 2]
```diff
@@ -102,3 +102,39 @@ roleRef:
...
+    app.kubernetes.io/component: events
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+subjects:
+  - kind: ServiceAccount
+    name: tekton-events-controller
+    namespace: tekton-pipelines
+roleRef:
+  kind: Role
+  name: tekton-pipelines-controller
```
Comment (wlynch) [inline config/201-rolebinding.yaml:121]: These are tied to the controller config options - not sure if you want the events controller to share the same config options as the pipeline controller or if you want to have it's own set of configs for flexibility.

File: config/201-rolebinding.yaml [hunk 3]
```diff
@@ -102,3 +102,39 @@ roleRef:
...
+    app.kubernetes.io/component: events
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+subjects:
+  - kind: ServiceAccount
+    name: tekton-events-controller
+    namespace: tekton-pipelines
+roleRef:
+  kind: Role
+  name: tekton-pipelines-leader-election
```
Comment (wlynch) [inline config/201-rolebinding.yaml:139]: I wonder if there's a way to isolate this set of leader election resources with the main controller 🤔  (i don't know if this is feasible)

File: config/events.yaml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    app.kubernetes.io/component: events
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/version: "devel"
+    app.kubernetes.io/part-of: tekton-pipelines
+    # tekton.dev/release value replaced with inputs.params.versionTag in pipeline/tekton/publish.yaml
+    pipeline.tekton.dev/release: "devel"
+    # labels below are related to istio and should not be used for resource lookup
+    app: tekton-pipelines-controller
+    version: "devel"
+  name: tekton-pipelines-controller
```
Comment (wlynch) [inline config/events.yaml:147]:
```suggestion
    app: tekton-events-controller
    version: "devel"
  name: tekton-events-controller
```

**Review by wlynch — CHANGES_REQUESTED** [review]

File: config/201-rolebinding.yaml (hunk 2 above)
Comment (afrittoli) [reply config/201-rolebinding.yaml:121]: My plan was for this PR to share the same configs, to keep the change as small as possible for users as well.
Once `TaskRun` and `PipelineRun` events are available in the new controller as well, I would start adding dedicated configurations.

File: config/201-rolebinding.yaml (hunk 3 above)
Comment (afrittoli) [reply config/201-rolebinding.yaml:139]: I *think* the leader's election of the new binary will be independent, even if they share the role, but I will double-check.

File: config/events.yaml (hunk 4 above)
Comment (afrittoli) [reply config/events.yaml:147]: Fixed

File: config/201-rolebinding.yaml (hunk 2 above)
Comment (wlynch) [reply config/201-rolebinding.yaml:121]: 👍 Fine with me, just wanted to make sure it was intentional 🙂

**Pull request merged**